### PR TITLE
Add hermit and update GH actions

### LINKED ISSUE #376: Hermitize
Setup instructions https://cashapp.github.io/hermit/

### PR DESCRIPTION
**Pull request by phoebe-lew**
closes #376

### CODE REVIEW

**Review by nitro-neal — APPROVED** [review]
awesome!

**Review by KendallWeihe — APPROVED** [review]

**Review by alecthomas — APPROVED** [review]

File: bin/activate-hermit [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+# This file must be used with "source bin/activate-hermit" from bash or zsh.
```
Comment (frankhinek) [inline bin/activate-hermit:2]: @phoebe-lew I guess this doesn't support `fish` shell but sounds like its possible to use:

https://cashapp.github.io/hermit/faq/#fish-support-via-direnv

File: .github/workflows/tests-ci.yml [hunk 2]
```diff
@@ -125,17 +125,10 @@ jobs:
...
 
-      - name: Set up Node.js
-        uses: actions/setup-node@5ef044f9d09786428e6e895be6be17937becee3a #v4.0.0
-        with:
-          node-version: 18
-          registry-url: https://registry.npmjs.org/
-          cache: 'pnpm'
+      # https://cashapp.github.io/hermit/usage/ci/
+      - name: Init Hermit
+        uses: cashapp/activate-hermit@v1
```
Comment (frankhinek) [inline .github/workflows/tests-ci.yml:131]: @phoebe-lew @shamilovtim Does caching still work with this approach?  Any impact to CI total runtime?

Comment (alecthomas) [reply .github/workflows/tests-ci.yml:131]: The cashapp/activate-hermit action does also support a `cache: true` attribute FYI.

Comment (shamilovtim) [reply .github/workflows/tests-ci.yml:131]: should that param be passed here?

Comment (shamilovtim) [reply .github/workflows/tests-ci.yml:131]: I see no difference in CI time though

Comment (phoebe-lew) [reply .github/workflows/tests-ci.yml:131]: @shamilovtim will add!

**Review by csuwildcat — APPROVED** [review]

**Review by mistermoe — APPROVED** [review]
Looks great. Only question is whether we want to keep node 18 as min or bump to 20

**Comment by phoebe-lew** [comment]
@mistermoe oops, I think we want to update to 20!

**Pull request merged**